specs/networking/p2p-interface.md

```diff
@@ -344,20 +344,20 @@ constituents individually as `response_chunk`s. For example, the
 Request, Response Content:
 ```
 (
-  head_fork_version: bytes4
-  finalized_root: bytes32
+  head_fork_version: Bytes4
+  finalized_root: Bytes32
   finalized_epoch: uint64
-  head_root: bytes32
+  head_root: Bytes32
   head_slot: uint64
 )
 ```
-The fields are:
+The fields are, as seen by the client at the time of sending the message:
 
 - `head_fork_version`: The beacon_state `Fork` version.
-- `finalized_root`: The latest finalized root the node knows about.
-- `finalized_epoch`: The latest finalized epoch the node knows about.
-- `head_root`: The block hash tree root corresponding to the head of the chain as seen by the sending node.
-- `head_slot`: The slot corresponding to the `head_root`.
+- `finalized_root`: The signing root of the latest finalized block.
+- `finalized_epoch`: The epoch of the block corresponding to `finalized_root`.
```

Comment:
This isn't exactly correct. The `finalized_epoch` and the `finalized_root` are best just grabbed from state.

There are scenarios in which `latest_finalized_epoch > slot_to_epoch(latest_finalized_block.slot)`

Reply:
Ah, might as well make it explicit, wdyt?

Reply:
...though technically that's a bit more restrictive, and clients can/should "optimize" it based on when `finalized_checkpoint` is updated.

Reply:
Do you mean update early if they know something is finalized, but the epoch transition hasn't yet happened?

About to make a mod to the fork choice in a separate PR that only allows updating the finalized info on the fork choice in a restricted range to avoid a type of "bounce attack" -- https://ethresear.ch/t/prevention-of-bouncing-attack-on-ffg/6114

Might affect this language here. we'll see

Reply:
hm, I guess perhaps that's possible if enough votes accumulate early (?), but I was more thinking about blocks during non-epoch slots - ie the finalized state of the latest epoch transition block is enough, you don't need the head block technically